src/api/src/resource/mod.rs

```diff
@@ -35,6 +35,18 @@ pub struct ResourceDesc {
     pub resource_tag: String,
 }
 
+impl ResourceDesc {
+    pub fn check_legality(&self) -> Result<()> {
+        if &self.repository_name == "." || &self.repository_name == ".." {
+            bail!(anyhow!("Illegal resource repository name"));
+        }
+        if &self.resource_type == "." || &self.resource_type == ".." {
```

Comment:
I assume we have already parsed out any slashes by this point?

Reply:
es, the Actix library will help us complete the parsing of slashes when parsing the path information of URL.